src/app/COMPONENTS/login-page/login-page.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, Inject, PLATFORM_ID } from '@angular/core'
+import { MaterialModule } from '../../material.module'
+import { FormBuilder, FormControl, FormGroup, ReactiveFormsModule,Validators } from '@angular/forms'
+import { CommonModule, isPlatformServer } from '@angular/common'
+import { UsersService } from '../../service/users/users.service'
+import { take, tap } from 'rxjs'
+import { Router } from '@angular/router'
+
+@Component({
+  selector: 'app-login-page',
+  standalone: true,
+  imports: [
+    MaterialModule,
+    CommonModule,
+    ReactiveFormsModule,
+  ],
+  templateUrl: './login-page.component.html',
+  styleUrl: './login-page.component.scss'
+})
+export class LoginPageComponent {
+
+  loginForm: FormGroup | null = null
+
+  isServer: boolean | null = null
+
+  constructor(private fb: FormBuilder, private userService: UsersService, private router: Router, @Inject(PLATFORM_ID) platformId: Object) {
+    this.isServer = isPlatformServer(platformId)
+
+    this.loginForm = this.fb.group({
+      username: new FormControl("", Validators.required),
+      password: new FormControl("", Validators.required),
+    })
+  }
+
+  signIn() {
+    if (this.loginForm?.valid) {
+      this.userService.signInUser(this.loginForm?.value).pipe(take(1), tap({
+        error: err => {
+          return this.loginForm?.setErrors({ unauthorised: true })
```

Comment:
you can move this to the subscribe part, since subscribe has both next and error